### PR TITLE
Fix Content Restriction & remove GetSettingsVariable binding

### PR DESCRIPTION
**Pull request by The-Aerec**
Co-Authored-By: Ophelia <heliashelium@gmail.com>

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Fixes Content_restrict setting in main settings

## Steps to test these changes

enable/disable content and then turn on content restriction

### CODE REVIEW

**Review by zach2good — COMMENTED** [review]
Do we use `GetSettingsVariable` in many places? Now that we have direct access to the settings (and there are more settings areas than just main) we can probably rip out this binding

**Comment by The-Aerec** [comment]
> Do we use GetSettingsVariable in many places? Now that we have direct access to the settings (and there are more settings areas than just main) we can probably rip out this binding

No not really. I didn't find many uses. This was just the quick fix Ophie and I found that fixed the issue.

Based on a quick search:

image

Yea its uh not used pretty much at all =p

**Comment by zach2good** [comment]
Replace then I think, then it's clear what file the setting is being read from, and it's easier to move them around in the future without breaking things

File: sql/npc_list.sql [hunk 1]
```diff
@@ -7628,7 +7628,7 @@ INSERT INTO `npc_list` VALUES (17093444,'Volker','Volker',0,0.000,0.000,0.000,32
 INSERT INTO `npc_list` VALUES (17093445,'Ajido-Marujido','Ajido-Marujido',0,0.000,0.000,0.000,32768,50,50,9,0,0,6,27,0x00004A0000000000000000000000000000000000,32,'TOAU',0);
 INSERT INTO `npc_list` VALUES (17093446,'Curilla','Curilla',0,0.000,0.000,0.000,32768,50,50,0,0,0,6,27,0x0000450000000000000000000000000000000000,32,'TOAU',0);
 INSERT INTO `npc_list` VALUES (17093447,'Ayame','Ayame',0,0.000,0.000,0.000,32768,50,50,0,1,0,6,27,0x0100030211100920003009400950006000700000,32,'TOAU',0);
-INSERT INTO `npc_list` VALUES (17093448,'Shantotto','Shantotto',0,0.000,0.000,0.000,32768,50,50,0,1,0,6,27,0x0100060600101320133013401350006000700000,32,'TOUA',0);
+INSERT INTO `npc_list` VALUES (17093448,'Shantotto','Shantotto',0,0.000,0.000,0.000,32768,50,50,0,1,0,6,27,0x0100060600101320133013401350006000700000,32,'TOAU',0);
```
Comment (zach2good) [inline sql/npc_list.sql:7631]: Nice catch

File: src/map/utils/zoneutils.cpp [hunk 2]
```diff
@@ -337,8 +337,8 @@ namespace zoneutils
 
     void LoadMOBList()
     {
-        uint8 normalLevelRangeMin = luautils::GetSettingsVariable("NORMAL_MOB_MAX_LEVEL_RANGE_MIN");
-        uint8 normalLevelRangeMax = luautils::GetSettingsVariable("NORMAL_MOB_MAX_LEVEL_RANGE_MAX");
+        uint8 normalLevelRangeMin = settings::get<int>("main.NORMAL_MOB_MAX_LEVEL_RANGE_MIN");
+        uint8 normalLevelRangeMax = settings::get<int>("main.NORMAL_MOB_MAX_LEVEL_RANGE_MAX");
```
Comment (zach2good) [inline src/map/utils/zoneutils.cpp]: Keep an eye on what types you're getting, this is a truncation: int -> uint8. 

I think for most of these settings that you've replaced, you should be asking for bool or uint8

File: src/map/lua/luautils.cpp [hunk 3]
```diff
@@ -1538,8 +1525,10 @@ namespace luautils
             }
             else
             {
+                std::string fileName("main.");
+                fileName.append(contentVariable);
                 // Cache contentTag lookups in a map so that we don't re-hit the Lua file every time
-                contentEnabled = (GetSettingsVariable(contentVariable.c_str()) != 0);
+                contentEnabled = (settings::get<int>(fileName)) != 0;
```
Comment (zach2good) [inline src/map/lua/luautils.cpp]: This section could be:
```cpp
contentEnabled = settings::get<bool>(fmt::format("main.{}", contentVariable));
```

File: src/map/lua/luautils.cpp [hunk 4]
```diff
@@ -251,7 +251,7 @@ namespace luautils
         }
 
         // Handle settings
-        contentRestrictionEnabled = GetSettingsVariable("RESTRICT_CONTENT") != 0;
+        contentRestrictionEnabled = settings::get<int>("main.RESTRICT_CONTENT") != 0;
```
Comment (zach2good) [inline src/map/lua/luautils.cpp]: contentRestrictionEnabled is a bool, ask settings for a bool: settings::get<bool>

If it's a 1 in the settings, it'll automagically convert it during lookup

File: src/map/battlefield.cpp [hunk 5]
```diff
@@ -258,7 +258,7 @@ void CBattlefield::ApplyLevelRestrictions(CCharEntity* PChar) const
         // Check if it's a mission and if config setting applies.
         if (!settings::get<bool>("map.LV_CAP_MISSION_BCNM") && m_isMission == 1)
         {
-            cap = luautils::GetSettingsVariable("MAX_LEVEL"); // Cap to server max level to strip buffs - this is the retail diff between uncapped and capped to max lv.
+            cap = settings::get<int>("main.MAX_LEVEL"); // Cap to server max level to strip buffs - this is the retail diff between uncapped and capped to max lv.
```
Comment (zach2good) [inline src/map/battlefield.cpp]: cap is uint8, ask settings for uint8

**Review by zach2good — CHANGES_REQUESTED** [review]
Almost there

File: src/map/lua/luautils.cpp [hunk 6]
```diff
@@ -1539,7 +1526,7 @@ namespace luautils
             else
             {
                 // Cache contentTag lookups in a map so that we don't re-hit the Lua file every time
-                contentEnabled = (GetSettingsVariable(contentVariable.c_str()) != 0);
+                contentEnabled = settings::get<bool>(fmt::format("main.{}", contentVariable)) != 0;
```
Comment (zach2good) [inline src/map/lua/luautils.cpp]: No need for ` != 0;`, this is now a bool of its own and doesn't need to converted into one

**Review by zach2good — CHANGES_REQUESTED** [review]

File: src/map/lua/luautils.cpp [hunk 7]
```diff
@@ -251,7 +251,7 @@ namespace luautils
         }
 
         // Handle settings
-        contentRestrictionEnabled = GetSettingsVariable("RESTRICT_CONTENT") != 0;
+        contentRestrictionEnabled = settings::get<bool>("main.RESTRICT_CONTENT") != 0;
```
Comment (zach2good) [inline src/map/lua/luautils.cpp]: Same here

**Review by zach2good — APPROVED** [review]
🎉

**Review by claywar — APPROVED** [review]

File: sql/npc_list.sql (hunk 1 above)
Comment (TeoTwawki) [reply sql/npc_list.sql:7631]: OH HOHO h...oh oh?

**Comment by TeoTwawki** [comment]
> Do we use `GetSettingsVariable` in many places? Now that we have direct access to the settings (and there are more settings areas than just main) we can probably rip out this binding

historical footnote: we used to use it for some more things that no longer exist. old old oooold times you'd have to go way back in history to see.

**Pull request merged**